go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -174,40 +201,84 @@ func (st *symtab) Find(col *sqlparser.ColName, autoResolve bool) (rb *route, isL
 	return rb, false, nil
 }
 
-func (st *symtab) searchColsyms(col *sqlparser.ColName) (rb *route, err error) {
-	var cursym *colsym
-	for _, colsym := range st.Colsyms {
-		if colsym.Alias.Equal(col.Name) {
+func (st *symtab) searchResultColumn(col *sqlparser.ColName) (rb *route, err error) {
+	var cursym *resultColumn
+	for _, rc := range st.ResultColumns {
+		if rc.alias.Equal(col.Name) {
 			if cursym != nil {
 				return nil, fmt.Errorf("ambiguous symbol reference: %v", sqlparser.String(col))
 			}
-			cursym = colsym
-			col.Metadata = colsym
+			cursym = rc
+			col.Metadata = rc.column
 		}
 	}
 	if cursym != nil {
-		return cursym.Route(), nil
+		return cursym.column.Route(), nil
 	}
 	return nil, nil
 }
 
 func (st *symtab) searchTables(col *sqlparser.ColName, autoResolve bool) *route {
+	// Identify the table.
+	var t *table
 	if col.Qualifier.IsEmpty() {
+		// if autoResolve is set, and there's only one table,
+		// we resolve to it.
 		if autoResolve && len(st.tables) == 1 {
-			col.Metadata = st.tables[0]
-			return st.tables[0].Route()
+			// Loop executes once.
```

Comment:
it looks like previously we took the first column, and in this approach it will take the last. Is that a problem?